### ISSUE TITLE
WhereYouGo no longer available for GARMIN Monterra

### ISSUE THREAD

**Issue by kurly1** [cgeo/WhereYouGo#156]
**Describe the bug:**
Since the last update of WherYouGo the app is no longer available in the Goolge Play Store for GARMIN Monterra devices. 

**Version of WhereYouGo used:**
The Version 0.9.3 was available in the Google Play store for the Monterra device.

**System information (Android version, device model):**
Monterra devices are still running on Ice Cream Sandwich (Android 4.0.4, API level 15).

**Additional context:**
I think this issue might be caused trou the increasing of the minSdkVersion from 15 to 16 in the build.gradle.

**Comment by Lineflyer** [comment]
> Monterra devices are still running on Ice Cream Sandwich (Android 4.0.4, API level 15).

Same problem as in c:geo.
Say goodbye to those unmaintained devices...its a real shame for Garmin.

**Comment by kurly1** [comment]
What was the reason that the supported API Level was raised from 15 to 16?

Because of the long battery run time this is still the best device to run Wherigo cartridges on.

Luckily I still have 0.9.3 on this device to run the cartridges.

**Comment by bekuno** [comment]
Many libraries are no more support api 15. And in the meantime the api 21 is now more realistic as minsdk for many apps and libs.

Maybe we can create a "legacy" version  version 0.9.4 with only changes to https.

**Comment by Lineflyer** [comment]
That should be no problem: We have a legacy branch, which is equal to 0.9.3. So we just need to push the https fix commit on top of that.The question is only if the configured builder on CI can still build it.

**Comment by bekuno** [comment]
@kurly1 
Please try this apk:
https://ci.cgeo.org/job/whereyougo%20(PR)/261/artifact/build/outputs/apk/debug/whereyougo-PR183-2020.05.24-e232935-debug.apk

**Comment by kurly1** [comment]
I installed this apk on my GARMIN Monterra and did the following tests:

- Intstallation went well.
- App is starting.
- Cardrige can be loaded.

I did not hat time to Play any Cardridge. And since c:geo is not available for this old Android version, I was not able to test the download of a cardridge from the WherYouGo app it selve. If any one can explain how to trigger a download manualy I will thest this as well.

**Comment by Lineflyer** [comment]
@kurly1 
As you seem to have a Garmin Monterra on Android 4.0 available:
We have now prepared an official legacy version to be published on Google Play.
As I do not have an Android 4.0 device but only 4.1 could you please do me the favor to install and test this one:
https://download.cgeo.org/whereyougo-legacy.apk
It should show the version 2020.06.25-legacy and download features should work.

**Comment by kurly1** [comment]
Exept the download Features the legacy Build seems to work. During the download I get a sslv3 handshake failure.

A download with the Chrome browser on this device is working. I don't know if the Chrome Comes with a own newer implementation of the SSL as provided by Android 4.0. It seems that there was a change in in the SSL implementation between Android 4.0 and 4.1.

@Lineflyer Dos the download feature work on Android 4.1?

**Comment by Lineflyer** [comment]
AFAIK Chrome has its own implementation, which is the reason why it sill works while the system itself fails. I am however wondering that it worked for you with the first debug version (see comments above) but not with the newly compiled version....mmh.

**Comment by jonas-koeritz** [comment]
Yep, this might actually work here as well. Maybe I've got that branch or code still around somewhere.

**Comment by Lineflyer** [comment]
@jonas-koeritz
AFAIR you proposed a change for cgeo to allow older devices to still connect (OKAPI). Do you think some similar workaround would be possible for this case here?

**Comment by kurly1** [comment]
I didn't test the download wit the first debug version as mentioned in the comments above, because I didn't know how I could test it. I never used the download function bevor.

Wit "Cardrige can be loaded" I ment that the Cardrige can be opened (loaded in to the app).

For me, the download functionality is not importend in the legacy version. It's better to have a version without a download functionality than have no version at all.

**Comment by Lineflyer** [comment]
@jonas-koeritz If you find time, you are welcome. I will just wait a bit with releasing a legacy version. Maybe we get the download working as well?!

**Comment by jonas-koeritz** [comment]
I think I will have a look at this...just need to get my thoughts on that
issue back together.

Lars <notifications@github.com> schrieb am Do., 2. Juli 2020, 21:01:

**Comment by Lineflyer** [comment]
@jonas-koeritz Shouldn't it be very similar to your pull request over at c:geo: https://github.com/cgeo/cgeo/pull/8139

**Comment by Lineflyer** [comment]
@jonas-koeritz Found any time meanwhile to take a look into this for the WhereYouGo legacy version (and quite similar for c:geo legacy version)?
